### PR TITLE
feat(ext/crypto): implement encrypt, decrypt & generateKey for RSA-OAEP

### PR DESCRIPTION
**Pull request by littledivy**
Towards #11690

### CODE REVIEW

File: cli/tests/unit/webcrypto_test.ts [hunk 1]
```diff
@@ -25,7 +25,7 @@ unitTest(async function testImportArrayBufferKey() {
 unitTest(async function testSignVerify() {
```
Comment (lucacasonato) [inline cli/tests/unit/webcrypto_test.ts:25]: Let's just duplicate the test. The conditional branching is confusing (especially with the name).

**Review by lucacasonato — COMMENTED** [review]
Implementation looks good from my side.

@bnoordhuis please review the Rust crypto code.

File: cli/tests/unit/webcrypto_test.ts [hunk 2]
```diff
@@ -70,6 +71,52 @@ unitTest(async function testSignVerify() {
...
+    const data = new Uint8Array([1, 2, 3]);
+    const encryptAlgorithm = { name: "RSA-OAEP" };
+    const encypted = await subtle.encrypt(
+      encryptAlgorithm,
+      keyPair.publicKey,
+      data,
+    );
+
+    assert(encypted);
+    assert(encypted.byteLength > 0);
```
Comment (bnoordhuis) [inline cli/tests/unit/webcrypto_test.ts:106]: For a 2048 bits key, that should always be 2048/8 - 42 = 214, right?

File: cli/tests/unit/webcrypto_test.ts [hunk 3]
```diff
@@ -70,6 +71,52 @@ unitTest(async function testSignVerify() {
...
+    const decrypted = await subtle.decrypt(
+      encryptAlgorithm,
+      keyPair.privateKey,
+      encypted,
+    );
+    assert(decrypted);
+    assert(decrypted instanceof ArrayBuffer);
+    assertEquals(new Uint8Array(decrypted), data);
+  }
+});
```
Comment (bnoordhuis) [inline cli/tests/unit/webcrypto_test.ts:118]: Can you also test encrypting > 214 bytes?

**Review by bnoordhuis — APPROVED** [review]

File: cli/tests/unit/webcrypto_test.ts (hunk 2 above)
Comment (littledivy) [reply cli/tests/unit/webcrypto_test.ts]: Yes. added a `encrypted.byteLength * 8 === 2048` assertion

File: cli/tests/unit/webcrypto_test.ts (hunk 3 above)
Comment (littledivy) [reply cli/tests/unit/webcrypto_test.ts:154]: Done

**Review by lucacasonato — APPROVED** [review]
LGTM!

**Pull request merged**